crypto/x509/x509_vfy.c

```diff
@@ -2980,7 +2984,14 @@ static int build_chain(X509_STORE_CTX *ctx)
                  * certificate among the ones from the trust store.
                  */
                 if ((search & S_DOALTERNATE) != 0) {
-                    OPENSSL_assert(num > i && i > 0 && ss == 0);
+                    if (!ossl_assert(num > i && i > 0 && ss == 0)) {
+                        X509err(X509_F_BUILD_CHAIN, ERR_R_INTERNAL_ERROR);
+                        X509_free(xtmp);
+                        trust = X509_TRUST_REJECTED;
+                        ctx->error = X509_V_ERR_UNSPECIFIED;
+                        search = 0;
+                        continue;
```

Comment:
In the context one can wonder if one should move `/* last chance to make a trusted chain */` block into the switch(trust) below. After all both ifs in this block are trust == X509_TRUST_UNTRUSTED, which would be different from X509_TRUST_REJECTED set by suggested assertion. Concern is following. With this modified control flow would it be appropriate to make sk_X509_num(ctx->chain) call?